app/controllers/orders_controller.rb

```diff
@@ -15,16 +15,16 @@ def index
     render json: @orders.to_json(proper_json)
   end
 
-  def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
     @order = Order.new(permitted_attributes.merge(created_by: current_user))
     authorize @order
 
     current_credit = @order.user&.credit
 
     if @order.save
-      if (@order.user&.provider == 'amber_oauth2') && @order.user&.credit&.negative? && current_credit&.positive?
+      if send_insufficient_credit_mail?(@order.user, current_credit)
         # User's credit went from positive to negative
-        UserCreditMailer.insufficient_credit_mail(current_user).deliver_later
+        UserCreditMailer.insufficient_credit_mail(@order.user).deliver_later
```

Comment:
We need to send the insufficient credit mail to the user that ordered, not the user that is logged in to the order screen